### PR TITLE
test: add tests for gulp function

### PR DESCRIPTION
**Pull request by 0xAustrian**
Closes BAL-56

### CODE REVIEW

File: test/unit/BPool.t.sol [hunk 1]
```diff
@@ -803,13 +809,50 @@ contract BPool_Unit_Unbind is BasePoolTest {
...
-  function test_Revert_NotBound() private view {}
+  struct Gulp_FuzzScenario {
+    address token;
+    uint256 balance;
+  }
 
-  function test_Revert_Reentrancy() private view {}
+  modifier happyPath(Gulp_FuzzScenario memory _fuzz) {
+    vm.assume(_fuzz.token != VM_ADDRESS);
+    vm.assume(_fuzz.token != 0x000000000000000000636F6e736F6c652e6c6f67);
```
Comment (0xJabberwock) [inline test/unit/BPool.t.sol:819]: What does `0x000000000000000000636F6e736F6c652e6c6f67` represent?

**Review by 0xJabberwock — APPROVED** [review]
✅

File: test/unit/BPool.t.sol (hunk 1 above)
Comment (0xAustrian) [reply test/unit/BPool.t.sol:819]: It's the console address, had to hardcode it since there is no accesible constant as `VM_ADDRESS`

Comment (0xJabberwock) [reply test/unit/BPool.t.sol:819]: Check https://github.com/foundry-rs/forge-std/pull/229.

**Pull request merged**